app/src/main/java/ch/epfl/sdp/blindly/recording/AudioLibraryAdapter.kt

```diff
@@ -0,0 +1,218 @@
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.annotation.RequiresApi
+import androidx.appcompat.widget.AppCompatImageButton
+import androidx.core.content.ContextCompat.startActivity
+import androidx.recyclerview.widget.RecyclerView
+import ch.epfl.sdp.blindly.R
+import ch.epfl.sdp.blindly.animations.RecordAnimations
+import ch.epfl.sdp.blindly.profile.ProfileFinished
+import java.io.File
+
+/**
+ * Serves as an adapter to add audio recordings in a RecyclerView
+ */
+class AudioLibraryAdapter(var recordList: ArrayList<AudioRecord>,
+                          var context: Context,
+                          private val listener: OnItemClickListener)
+    : RecyclerView.Adapter<AudioLibraryAdapter.ViewHolder>() {
+    /**
+     * Provide a reference to the type of views that you are using
+     * (custom ViewHolder).
+     */
+    var mediaPlayer: MediaPlayer? = null
+    private var isPlayerPaused = false
+    private var isPlayerStopped = true
+
+    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener {
+        val recordName: TextView = view.findViewById(R.id.recordName)
+        val recordDuration: TextView = view.findViewById(R.id.recordDuration)
+        val nameDurationLayout: LinearLayout = view.findViewById(R.id.nameDurationLayout)
+        val expandableLayout: RelativeLayout = view.findViewById(R.id.audioPlayLayout)
+        val playPauseButton: AppCompatImageButton = view.findViewById(R.id.playPauseButton)
+        val playBar: SeekBar = view.findViewById(R.id.playBar)
+        val playTimer: Chronometer = view.findViewById(R.id.audioTimer)
+        val remainingTimer: Chronometer = view.findViewById(R.id.remainingTimer)
+        val selectButton: Button = view.findViewById(R.id.selectButton)
+
+        init {
+            // Define click listener for the ViewHolder's View.
+            nameDurationLayout.setOnClickListener(this)
+            expandableLayout.visibility = View.GONE
+        }
+
+        override fun onClick(v: View?) {
+            // Forward the click to the main activity
+            val position = adapterPosition
+            if (position != RecyclerView.NO_POSITION) {
+                listener.onItemClick(position)
+            }
+        }
+    }
+
+    // Expands or collapse the layout for playing the audio file
+    private fun toggleLayout(isExpanded: Boolean, v: View, layoutExpand: RelativeLayout): Boolean {
+        if (isExpanded) {
+            RecordAnimations.expand(layoutExpand)
+        } else {
+            RecordAnimations.collapse(layoutExpand)
+        }
+        return isExpanded
+    }
+
+    // Create new views (invoked by the layout manager)
+    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
+        // Create a new view, which defines the UI of the list item
+        val view = LayoutInflater.from(context).inflate(R.layout.audio_recording, null)
```

Comment:
Cirrus doesn't seem too happy about the root being null, can't you pass something else as the ViewGroup root ?

Reply:
I've searched on the web and I think we can leave it as it is, because the parent is not known there.